components/webgpu/wgpu_thread.rs

```diff
@@ -868,21 +872,70 @@ impl WGPU {
                             );
                         };
                     },
+                    WebGPURequest::BeginRenderPass {
+                        command_encoder_id,
+                        render_pass_id,
+                        label,
+                        color_attachments,
+                        depth_stencil_attachment,
+                        device_id: _device_id,
+                    } => {
+                        let global = &self.global;
+                        let desc = &RenderPassDescriptor {
+                            label,
+                            color_attachments: color_attachments.into(),
+                            depth_stencil_attachment: depth_stencil_attachment.as_ref(),
+                            timestamp_writes: None,
+                            occlusion_query_set: None,
+                        };
+                        let (pass, error) = gfx_select!(
+                            command_encoder_id => global.command_encoder_create_render_pass_dyn(
+                                command_encoder_id,
+                                desc,
+                        ));
+                        assert!(
+                            self.render_passes
+                                .insert(render_pass_id, (pass, error.is_none()))
+                                .is_none(),
+                            "RenderPass should not exist yet."
+                        );
+                        // TODO: Command encoder state errors
+                        // self.maybe_dispatch_wgpu_error(device_id, error);
+                    },
+                    WebGPURequest::RenderPassCommand {
+                        render_pass_id,
+                        render_command,
+                        device_id,
+                    } => {
+                        if let Some((pass, valid)) = self.render_passes.get_mut(&render_pass_id) {
```

Comment:
Why not only remove on drop, and rely on the error returned by wgpu? It should return an error if the pass has already ended. Since we are relying on wgpu for all other logic, I don't see why we should keep track of the ended status of the pass by removing it ourselves from the map.

Reply:
Like for compute pass, we need to drop the pass when it ends.

Reply:
Ok